flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinatorRestoringTest.java

```diff
@@ -1213,31 +1213,153 @@ public void testRestoringPartiallyFinishedChainsFails() throws Exception {
                         + "anon("
                         + jobVertexID1
                         + ")"
-                        + " which contain both finished and unfinished operators");
+                        + " which contain mixed operator finished state: [ALL_RUNNING, FULLY_FINISHED]");
         coord.restoreLatestCheckpointedStateToAll(vertices, false);
     }
 
     @Test
     public void testAddingRunningOperatorBeforeFinishedOneFails() throws Exception {
```

Comment:
Could we also have a test case for the situation that a single operator is connected via two different distribution patterns? That is quite an uncommon and very specific scenario that in my mind is worth checking.

Reply:
A test should be needed. I added the test~